src/main/java/io/github/microcks/testcontainers/MicrocksContainer.java

```diff
@@ -561,6 +564,115 @@ public void downloadAsSecondaryRemoteArtifact(String remoteArtifactUrl) throws A
+   }
+
+   private boolean doVerify(String serviceName, String serviceVersion, Date invocationsDate) {
+      DailyInvocationStatistic dailyInvocationStatistic = getServiceInvocations(getHttpEndpoint(), serviceName, serviceVersion, invocationsDate);
+
+      if (dailyInvocationStatistic == null) {
+         return false;
+      }
+
+      BigDecimal count = dailyInvocationStatistic.getDailyCount();
+
+      return count != null && count.intValue() != 0;
+   }
+
+   /**
+    * Get the invocations' count for a given service, identified by its name and version, for the current invocations' date.
+    *
+    * @param serviceName    mandatory
+    * @param serviceVersion mandatory
+    * @return zero if the given service was not found, or has never been invoked. Else, returns the daily count of invocations for the given service.
+    */
+   public Long getServiceInvocationsCount(String serviceName, String serviceVersion) {
+      return this.doGetServiceInvocationsCount(serviceName, serviceVersion, null);
+   }
+
+   /**
+    * Get the invocations' count for a given service, identified by its name and version, for the given invocations' date.
+    * In unit testing context, it should be useless to pass the invocations date, prefer calling {@link #getServiceInvocationsCount(String, String)}`.
+    *
+    * @param serviceName     mandatory
+    * @param serviceVersion  mandatory
+    * @param invocationsDate nullable
+    * @return zero if the given service was not found, or has never been invoked. Else, returns the daily count of invocations for the given service.
+    */
+   public Long getServiceInvocationsCount(String serviceName, String serviceVersion, Date invocationsDate) {
+      return this.doGetServiceInvocationsCount(serviceName, serviceVersion, invocationsDate);
+   }
+
+   private Long doGetServiceInvocationsCount(String serviceName, String serviceVersion, Date invocationsDate) {
+      DailyInvocationStatistic dailyInvocationStatistic = getServiceInvocations(getHttpEndpoint(), serviceName, serviceVersion, invocationsDate);
+
+      if (dailyInvocationStatistic == null) {
+         return 0L;
+      }
+
+      BigDecimal count = dailyInvocationStatistic.getDailyCount();
+
+      return count.longValue();
+   }
+
+
+   /**
+    * Returns all data from Microcks Metrics REST API about the invocations of a given service.
+    *
+    * @param microcksContainerHttpEndpoint mandatory
+    * @param serviceName                   mandatory
+    * @param serviceVersion                mandatory
+    * @param invocationsDate               nullable
+    * @return
+    */
+   private static DailyInvocationStatistic getServiceInvocations(String microcksContainerHttpEndpoint, String serviceName, String serviceVersion, Date invocationsDate) {
+      String encodedServiceName = UrlEscapers.urlFragmentEscaper().escape(serviceName);
+      String encodedServiceVersion = UrlEscapers.urlFragmentEscaper().escape(serviceVersion);
```

Comment:
I'm not sure this is needed as most of the HTTP client libraries or utilities are doing the encoding of the URL. Could you please check this one? If actually needed, couldn't we use instead the standard `URLEncoder.encode()` to avoid depending on third-party libs?

Reply:
I encountered space character encoding issue.
Using `URLEncoder.encode()`, with `UTF-8`, it is changed in "%20".
Whereas the Metrics REST API expect the space character to be changed to "+".
Maybe there is another way to encode the service name so as it is correctly sent to the Metrics REST API ?

Reply:
I think `+` and `%20` are both accepted. I will double-check this point.